src/main/java/org/springframework/cloud/deployer/spi/cloudfoundry/CloudFoundryAppDeployer.java

```diff
@@ -158,12 +160,12 @@ public AppStatus status(String id) {
 		return Flux.fromStream(request.getDefinition().getProperties().entrySet().stream());
 	}
 
-	/**
-	 * TODO: Should we join properties with override, or replace?
-	 *
-	 * @param request
-	 * @return
-	 */
+	private String deploymentId(AppDeploymentRequest request) {
+		return String.format("%s-%s",
+				request.getEnvironmentProperties().get(GROUP_PROPERTY_KEY),
```

Comment:
Comes from the caller, as a well known AppDeployer property. See [here](https://github.com/spring-cloud/spring-cloud-deployer/blob/6e44c8ed742a890cf2131c79ca61bb6af74a7513/spring-cloud-deployer-spi/src/main/java/org/springframework/cloud/deployer/spi/app/AppDeployer.java#L39-L42). Not very clear though if it is mandatory. cc @markfisher